tools/gfx/render.h

```diff
@@ -1113,37 +1342,36 @@ class IRenderer: public ISlangUnknown
     }
 
 // ----------------------------------------------------------------------------------------
-inline void IRenderer::setVertexBuffer(UInt slot, BufferResource* buffer, UInt stride, UInt offset)
+inline void IRenderer::setVertexBuffer(UInt slot, IBufferResource* buffer, UInt stride, UInt offset)
 {
     setVertexBuffers(slot, 1, &buffer, &stride, &offset);
 }
 
-/// Functions that are around Renderer and it's types
-struct RendererUtil
+// Global public functions
+
+extern "C"
 {
-    typedef SlangResult (*CreateFunc)(IRenderer** outRenderer);
+    typedef SlangResult(SLANG_MCALL * SGRendererCreateFunc)(IRenderer** outRenderer);
 
-        /// Gets the size in bytes of a Format type. Returns 0 if a size is not defined/invalid
-    SLANG_FORCE_INLINE static size_t getFormatSize(Format format) { return s_formatSize[int(format)]; }
-        /// Given a renderer type, gets a projection style
-    static ProjectionStyle getProjectionStyle(RendererType type);
+    /// Gets the size in bytes of a Format type. Returns 0 if a size is not defined/invalid
+    SLANG_GFX_API size_t SLANG_MCALL gfxGetFormatSize(Format format);
 
-        /// Given the projection style returns an 'identity' matrix, which ensures x,y mapping to pixels is the same on all targets
-    static void getIdentityProjection(ProjectionStyle style, float projMatrix[16]);
+    /// Gets the binding style from the type
+    SLANG_GFX_API BindingStyle SLANG_MCALL gfxGetBindingStyle(RendererType type);
 
-        /// Get the binding style from the type
-    static BindingStyle getBindingStyle(RendererType type) { return s_rendererTypeToBindingStyle[int(type)]; }
+    /// Given a renderer type, gets a projection style
+    SLANG_GFX_API ProjectionStyle SLANG_MCALL gfxGetProjectionStyle(RendererType type);
```

Comment:
I am not a fan of these functions, and I'm not convinced we ever should have added them. If we can find a way to remove them as part of the future refactoring work, that would be ideal.